packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -417,7 +417,20 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       // Normally we can bail out on props equality but if context has changed
       // we don't do the bailout and we have to reuse existing props instead.
     } else if (memoizedProps === nextProps) {
-      return bailoutOnAlreadyFinishedWork(current, workInProgress);
+      if (
+        workInProgress.mode & AsyncMode &&
+        shouldDeprioritizeSubtree(type, nextProps)
+      ) {
+        if (renderExpirationTime === Never) {
```

Comment:
This condition should be first because it is the fastest one. No property access.

Reply:
Ha, I was trying to avoid assigning the result of `shouldDeprioritizeSubtree` to a boolean :D